src/messages.rs

```diff
@@ -179,34 +201,61 @@ pub struct Header {
 }
 
 impl Serializable for Header {
-    fn to_bytes(&self) -> Vec<u8> {
-        let mut buf = BytesMut::with_capacity(300);
-        buf.put_u8(self.topic);
-        buf.put_u8(self.step);
-        buf.put_u64_le(self.round);
-        buf.put(&self.block_hash[..]);
-        buf.put(&self.pubkey_bls.bytes()[..]);
-
-        buf.to_vec()
-    }
-
-    fn from_bytes(buf: &mut Bytes) -> Self {
-        let mut header = Header {
-            topic: buf.get_u8(),
-            step: buf.get_u8(),
-            round: buf.get_u64_le(),
-            ..Default::default()
-        };
-        buf.copy_to_slice(&mut header.block_hash[..]);
+    fn write<W: Write>(&self, w: &mut W) -> io::Result<()> {
+        w.write_all(&[self.topic])?;
+        Self::write_var_le_bytes(w, &self.pubkey_bls.bytes()[..])?;
+        w.write_all(&self.round.to_le_bytes())?;
+        w.write_all(&[self.step])?;
+        w.write_all(&self.block_hash[..])?;
+
+        Ok(())
+    }
+
+    /// Deserialize struct from buf by consuming N bytes.
+    fn read<R: Read>(r: &mut R) -> io::Result<Self>
+    where
+        Self: Sized,
+    {
+        // Read topic
+        let mut buf = [0u8; 1];
+        r.read_exact(&mut buf)?;
+        let topic = buf[0];
+
+        // Read bls pubkey
+        let buf: [u8; 96] = Self::read_var_le_bytes(r)?;
```

Comment:
Do we need a var_le_bytes even if we know the exact size of the field?

Reply:
Unfortunately yes, we need it. This is again to be compatible with wire protocol (golang) implementation.